src/hypothesis/strategies.py

```diff
@@ -901,7 +901,15 @@ def accept(*args, **kwargs):
         class CompositeStrategy(SearchStrategy):
 
             def do_draw(self, data):
-                return f(data.draw, *args, **kwargs)
+                first_draw = [True]
+
+                def draw(strategy):
+                    if not first_draw[0]:
+                        data.mark_bind()
+                    first_draw[0] = False
```

Comment:
Because it gets updated inside the draw() function, and because Python 2 compatibility we can't declare it nonlocal.